build.xml

```diff
@@ -1,126 +1,417 @@
+        </sequential>
+    </macrodef>
+
+    <condition property="is.atLeastJava9" value="true" else="false">
+        <or>
+            <matches string="${java.version}" pattern="^9"/>
+            <matches string="${java.version}" pattern="^1[0-9]"/>
+        </or>
+    </condition>
+    <condition property="kernel" value="darwin-amd64" else="linux-amd64">
+        <os family="mac"/>
+    </condition>
+    <condition property="home.ext" value="/Contents/Home" else="">
+        <os family="mac"/>
+    </condition>
+
+    <travis target="env" start="Environment" />
+    <echo>
+        ant.java.version: ${ant.java.version}
+        java.version:     ${java.version}
+        is.atLeastJava9:  ${is.atLeastJava9}
+        kernel:           ${kernel}
+        env.TRAVIS:       ${env.TRAVIS}
+    </echo>
+    <travis target="env" />
 
     <property name="src.dir"     value="src"/>
-    <property name="test.dir"    value="tests"/>
-    <property name="lib.dir"     value="libs" />
+    <property name="src_gen.dir" value="src_gen"/>
+    <property name="lib.dir"     location="libs" />
+    <property name="unit.dir"    value="tests/som" />
+    <property name="corelib.dir" location="core-lib" />
+    <property name="graal.dir"   location="${lib.dir}/truffle/compiler" />
+    <property name="truffle.dir" location="${lib.dir}/truffle/truffle" />
+    <property name="sdk.dir"     location="${lib.dir}/truffle/sdk" />
+    <property name="tools.dir"   location="${lib.dir}/truffle/tools" />
+    <property name="svm.dir"     location="${lib.dir}/truffle/substratevm" />
+    <property name="compiler.build" location="${graal.dir}/mxbuild/dists/jdk1.8" />
+    <property name="sdk.build"   location="${sdk.dir}/mxbuild/dists/jdk1.8" />
+    <property name="tools.build" location="${tools.dir}/mxbuild/dists/jdk1.8" />
+    <property name="svm.build"   location="${svm.dir}/mxbuild/dists/jdk1.8" />
+    <property name="truffle.build" location="${truffle.dir}/mxbuild/dists/jdk1.8" />
+    <property name="checkstyle.version" value="8.28" />
+    <property name="jacoco.version"     value="0.8.5" />
+    <property name="jvmci.home"  location="${lib.dir}/jvmci${home.ext}" />
+    <property name="junit.version" value="4.12" />
     <property name="build.dir"   value="build"/>
     <property name="classes.dir" value="${build.dir}/classes"/>
-    <property name="junit.version" value="4.12" />
 
-    <property name="checkstyle.version" value="8.14" />
+    <property name="mx.cmd" value="../../mx/mx" />
 
     <property environment="env"/>
 
-    <path id="project.classpath">
+<!--    <property name="src.dir"     value="src"/>-->
+<!--    <property name="test.dir"    value="tests"/>-->
+<!--    <property name="lib.dir"     value="libs" />-->
+<!--    <property name="build.dir"   value="build"/>-->
+<!--    <property name="classes.dir" value="${build.dir}/classes"/>-->
+
+
+    <property name="checkstyle.version" value="8.14" />
+
+<!--    <path id="project.classpath">-->
```

Comment:
Same here, I'd remove all old code. We have other places where we keep it for reference.